modules.d/90dmsquash-live/dmsquash-live-root.sh

```diff
@@ -286,28 +287,49 @@ fi
 if [ -e /run/initramfs/live/${live_dir}/${squash_image} ]; then
     SQUASHED="/run/initramfs/live/${live_dir}/${squash_image}"
 fi
+if [ -e /run/initramfs/live/${live_dir}/${join} ]; then
+    SQUASHED_JOIN="/run/initramfs/live/${live_dir}/${join}"
+fi
 if [ -e "$SQUASHED" ]; then
     if [ -n "$live_ram" ]; then
         echo 'Copying live image to RAM...' > /dev/kmsg
         echo ' (this may take a minute)' > /dev/kmsg
         dd if=$SQUASHED of=/run/initramfs/squashed.img bs=512 2> /dev/null
         echo 'Done copying live image to RAM.' > /dev/kmsg
         SQUASHED="/run/initramfs/squashed.img"
+        if [ -n ${join} ];then
+            echo 'Copying live image to RAM...' > /dev/kmsg
+            echo ' (this may take a minute)' > /dev/kmsg
+            dd if=$SQUASHED_JOIN of=/run/initramfs/${join} bs=512 2> /dev/null
+            echo 'Done copying live image to RAM.' > /dev/kmsg
+            SQUASHED_JOIN="/run/initramfs/${join}"
+        fi
     fi
 
     SQUASHED_LOOPDEV=$( losetup -f )
     losetup -r $SQUASHED_LOOPDEV $SQUASHED
     mkdir -m 0755 -p /run/initramfs/squashfs
     mount -n -t squashfs -o ro $SQUASHED_LOOPDEV /run/initramfs/squashfs
 
+    if [ -n ${join} ]; then
+        SQUASHED_JOIN_LOOPDEV=$( losetup -f )
+        losetup -r $SQUASHED_JOIN_LOOPDEV $SQUASHED_JOIN
+        mkdir -m 0755 -p /run/initramfs/joinfs
+        mount -n -t squashfs -o ro $SQUASHED_LOOPDEV /run/initramfs/joinfs
```

Comment:
```suggestion
        mount -n -t squashfs -o ro $SQUASHED_JOIN_LOOPDEV /run/initramfs/joinfs
```
Did you intend to support an unflat join.img?  I would recommend not to as you require an OverlayFS for the union join.  (You don't use the /run/initramfs/joinfs mountpoint anyway.  This section is superfluous.)